src/DynamicObj/DynObj.fs

```diff
@@ -63,39 +65,128 @@ module DynObj =
+    /// <param name="dynObj">The DynamicObj to set the property on</param>
+    /// <remarks>This function mutates the input DynamicObj</remarks>
+    let withValue (propertyName:string) (propertyValue:'TPropertyValue) (dynObj: DynamicObj) =
+        setValue propertyName propertyValue dynObj
+        dynObj
+
+    /// <summary>
+    /// Sets the given dynamic property name and value on the given DynamicObj if the value is Some('TPropertyValue).
+    /// If the given propertyValue is None, does nothing to the input DynamicObj.
+    /// </summary>
+    /// <param name="propertyName">The name of the dynamic property to set</param>
+    /// <param name="propertyValue">The value of the dynamic property to set</param>
+    /// <param name="dynObj">The DynamicObj to set the property on</param>
+    /// <remarks>This function mutates the input DynamicObj</remarks>
+    let setValueOpt (propertyName: string) (propertyValue: 'TPropertyValue option) (dynObj: DynamicObj) = 
+        match propertyValue with
+        | Some pv -> dynObj |> setValue propertyName pv
         | None -> ()
 
-    let setValueOptBy (dyn:DynamicObj) propName f = 
-        function
-        | Some o -> 
-            dyn.SetValue(propName,f o)
+    /// <summary>
+    /// Sets the given dynamic property name and value on the given DynamicObj if the value is Some('TPropertyValue) and returns it.
+    /// If the given propertyValue is None, returns the unchanged DynamicObj.
+    /// </summary>
+    /// <param name="propertyName">The name of the dynamic property to set</param>
+    /// <param name="propertyValue">The value of the dynamic property to set</param>
+    /// <param name="dynObj">The DynamicObj to set the property on</param>
+    /// <remarks>This function mutates the input DynamicObj</remarks>
+    let withValueOpt (propertyName: string) (propertyValue: 'TPropertyValue option) (dynObj: DynamicObj) = 
+        match propertyValue with
+        | Some pv -> dynObj |> withValue propertyName pv
+        | None -> dynObj
+
+    /// <summary>
+    /// Sets the given dynamic property name with the result of a mapping function applied to the given property value on the given DynamicObj if the value is Some('TPropertyValue).
+    /// If the given propertyValue is None, does nothing to the input DynamicObj.
+    /// </summary>
+    /// <param name="propertyName">The name of the dynamic property to set</param>
+    /// <param name="propertyValue">The value of the dynamic property to set</param>
+    /// <param name="mapping">A function to apply to the property value before setting it on the DynamicObj</param>
+    /// <param name="dynObj">The DynamicObj to set the property on</param>
+    /// <remarks>This function mutates the input DynamicObj</remarks>
+    let setValueOptBy (propertyName: string) (propertyValue: 'TPropertyValue option) (mapping: 'TPropertyValue -> 'UPropertyValue) (dynObj: DynamicObj) = 
```

Comment:
This seems oddly specific

Reply:
and used hundreds of times in Plotly.NET